### ISSUE TITLE
Uses MD5 for Authentication

### ISSUE THREAD

**Issue by AlexBestoso** [dotproject/dotProject#73]
I was playing with the source code, and noticed that the php login uses an unsalted MD5 hashing function. The issue was easy to fix, though I felt like you should be aware. 

Replacing md5 hashing functions with hash('sha512', $password, FALSE) seems to have worked for me. After changing user_password entries in the db, and looking through the php files.

**Comment by ajdonnison** [comment]
Given there will be existing sites out there we need to add a little to handle the transition.  I agree we need to update the hash function, I don't agree we force sites to update all their passwords manually.

**Comment by AlexBestoso** [comment]
I agree. 
Perhaps a system that has users type in their user and password into the login form. 

On post, query for user_username. If the username exists, check if user_password is _not_ the length of a SHA512 sum. 

If _not_ the length of a SHA512 sum, hash the password with MD5 and authenticate, else hash with SHA512. 

If the user authenticates with MD5. Hash the original password again, but this time using a stronger algorithm. Then update the SQL database entry where user_username="form input".

SHA-1 has a collision vulnerability that was discovered in 2017, so that one can't be used.
The idea needs to be refined to ensure no passwords or hashes are accidentally leaked.

**Comment by ajdonnison** [comment]
Yep, that is pretty much the way I think it should go too.  Are you up for creating a pull request for that?

**Comment by AlexBestoso** [comment]
Sure!
